### ISSUE TITLE
Inconsitent tab usage confusing to screen readers

### ISSUE THREAD

**Issue by kimisgold** [omeka/Omeka#979]
Our accessibility audit pointed out that switching between tabs within multiple sections of the admin area brings focus to the top of the page. In these instances, each tab is actually its own view, as opposed to being a single view with tabs exposing different sections like in the item metadata editor.

A sample of areas with the troublesome pattern:
* Appearance
* Settings
* Contribution plugin views

**Comment by zerocrates** [comment]
Yes, we have several spots like those where the tabs are just visual.

**Comment by kimisgold** [comment]
What's the most relevant WCAG requirement for this issue?

**Comment by aviannamiller** [comment]
I would say it's an issue mostly in regards to [SC 3.2.4 Consistent Identification (Level AA)](https://w3.org/TR/WCAG21#consistent-identification) with elements of [SC 2.4.3 Focus Order (Level A)](https://w3.org/TR/WCAG22#focus-order) (and by extension [SC 1.3.2 Meaningful Sequence (Level A)](https://w3.org/TR/WCAG21#meaningful-sequence)). The expected result based on the visual pattern and way it's implemented elsewhere is for a dynamic change that continues the reading order / sequence by moving focus onto the next element, but since the page loads to a new view, focus is reset to the top. As Kim pointed out here, introducing a different appearance for this might be the way towards the best solution so as not to continue the issue of similarly identified patterns not producing inconsistent results for users.

**Comment by kimisgold** [comment]
Maybe we make the distinction that any view currently presented as a tab and requiring a page load appears as a child page in the content navigation on the left? This seems to mainly apply to views belonging to the global nav, and would likely involve a relocation.

<img width="564" height="338" alt="Draft of Omeka Classic interface, showing active global section in the left navigation." src="https://github.com/user-attachments/assets/16490276-e953-4d63-a098-48158f2790a1" />

**Comment by aviannamiller** [comment]
That sounds great to me. If we can do it with a breadcrumbs nav in or around the main content area as well, even better.

**Comment by kimisgold** [comment]
Experimenting with that pattern, it's maybe too big of a navigation redesign for the moment. I'm starting out with a simpler approach in the [`a11y/fix-tabs`](https://github.com/omeka/Omeka/tree/a11y/fix-tabs) branch. In this branch, any section navigation using links leading to page loads are now presented as rows of plain links. Views using tabs now have revised markup using the appropriate tab roles and buttons instead of anchors. I've also restyled the tabs to have more of a condensed pill style that wraps better than previously. Still mulling on how I want to present breadcrumbs.

<img width="693" height="242" alt="Screenshot depicting new sectio nav style for Appearance section" src="https://github.com/user-attachments/assets/a333b84e-5eed-4d9b-b167-746125de1755" />

<img width="698" height="242" alt="Screenshot depicting new tab style for item form" src="https://github.com/user-attachments/assets/0eb269c9-9d39-4c97-9dce-4c1ce1c1967b" />

**Comment by aviannamiller** [comment]
I really like the redesigned tabs. However, I think something may not be meshing with the `ul` combined with the tab pattern, potentially causing some of the roles involved to conflict.

Right when the page loads, this flashes for a second before the tab pattern fully loads as expected:

<img width="1230" height="366" alt="Detail of Collections page. Navigation tabs appear on page in white rectangular container" src="https://github.com/user-attachments/assets/0f66924e-6902-4701-8de5-66d36e3d8333" />

If the links are still located in the main content area, as you've done here, breadcrumbs seems unnecessary. If down the road you move it to the sidebar menu, then that would definitely be a request again. With that said, the current implementation should have a label on the nav element to better speak to the context of the nav links.

In instances where this has carried over to plugins, the links appear in an un-styled ul:

<img width="731" height="188" alt="Detail of links nav region on Simple Pages page. Links are arranged vertically in an unordered list." src="https://github.com/user-attachments/assets/13fbda5a-2c89-4060-82e1-d496b819168b" />

**Comment by kimisgold** [comment]
Ah, the section nav and form tabs were being rendered inside the content area then moved outside of it using JS. That should now be fixed across the views.

**Comment by aviannamiller** [comment]
That seems better.

However, I'm no longer seeing the Show Details button or `#batch-all-checkbox` on the items browse view, not entirely positive if anything else has been impacted, but that's what I'm immediately noticing upon comparison with `master`.

**Comment by kimisgold** [comment]
I had forgotten to remove a callback to removed js moving the nav list, so it broke the JS a bit. That should be fixed now.

**Comment by aviannamiller** [comment]
Okay, yes, this portion looks to be fixed.

**Comment by kimisgold** [comment]
On reflection, I realize the moveNavList js needs to stay in for the sake of plugins that haven't updated yet. That said, plugins like Contribution and CSV Import carry over fairly gracefully. Simple Pages uses a pattern in consistent with the others, so it needs to be updated.

What else is left in this issue? The tab lists and `<nav>` elements in the core should all have labels.

**Comment by aviannamiller** [comment]
Is Local Contexts in the same boat as Simple Pages?

In terms of the core, the `<nav>` elements in Settings and Users edit seem to be missing labels. The only instance I'm seeing of a labeled `nav` is in Appearance.

I'm still seeing some errors regarding the roles in the tab pattern, not sure how relevant some of these are:

- From the validator:
  - Every active `role=tab` element must have a corresponding `role=tabpanel` element.
          `<li><button type="button" role="tab" aria-selected="true" aria-controls="dublin-core-metadata" class="active">`

- Accessibility Insights:
  - WCAG 1.3.1: Certain ARIA roles must contain particular children (`#section-tabs`)
  Snippet: `<ul id="section-tabs" aria-label="Form" class="tabs navigation" role="tablist">`
  - WCAG 1.3.1: Certain ARIA roles must be contained by particular parents (`button[aria-controls="dublin-core-metadata"]`)
  Snippet: `<button type="button" role="tab" aria-selected="false" aria-controls="dublin-core-metadata">Dublin Core</button>`
  - WCAG 1.3.1: `<li>` elements must be contained in a `<ul>` or `<ol>` (`li:nth-child(2)`)
  Snippet: `<li><button type="button" role="tab" aria-selected="false" aria-controls="dublin-core-metadata">Dublin Core</button></li>`

**Comment by kimisgold** [comment]
Gotcha, the tab pattern should be updated to provide the missing attributes. The Settings and Users nav should now have `aria-label`s as well.

**Comment by aviannamiller** [comment]
I'm now experiencing some skipping of tabs when navigating via the tab key. Moving via arrow keys works fine.

Also, on the Users edit pages, some layout bugs:

<img width="1422" height="636" alt="User edit page in admin. Labels within edit form are pushed to the left, partially hidden under menu sidebar." src="https://github.com/user-attachments/assets/7718a889-543b-4212-84e1-d4559465f02f" />

**Comment by aviannamiller** [comment]
Further tab key testing on the tab pattern in item edit: VoiceOver seems to skip tabs more arbitrarily, while JAWS stops on the first tab, then moves into the form. NVDA allows navigation through a number of tabs, then moves into the form, skipping the last few.

**Comment by kimisgold** [comment]
The Users nav had a missed closing tag, which should now be fixed.

Regarding tabpanel keyboard navigation, my understanding is that arrow keys should replace the tab key in how the user browses through each item. It's based on the behaviors in this [WCAG example of manually activated tabs](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/). I'm going to look into those screen reader oddities, but in the meantime, can you verify that you're not seeing similar behaviors in linked example?

**Comment by aviannamiller** [comment]
Users nav looks good.

I see, so it's similar to the behavior of navigating a group of radio controls. The WCAG example not have seem to have that problem, any tested screen readers or browsers. I'm also still seeing some of the same warnings when I check with Accessibility Insights:

- WCAG 1.3.1: Certain ARIA roles must contain particular children (`#section-tabs`)
Snippet: `<ul id="section-tabs" aria-label="Form" class="tabs navigation" role="tablist">`
- WCAG 1.3.1: Certain ARIA roles must be contained by particular parents (`button[aria-controls="dublin-core-metadata"]`)
Snippet: `<button type="button" tabindex="-1" role="tab" aria-selected="false" aria-controls="dublin-core-metadata">Dublin Core</button>`
- WCAG 1.3.1: `<li>` elements must be contained in a `<ul>` or `<ol>` (`li:nth-child(2)`)
Snippet: `<li><button type="button" tabindex="-1" role="tab" aria-selected="false" aria-controls="dublin-core-metadata">Dublin Core</button></li>`

Could the `<ul>` be causing some of this unexpected screen reader behavior?

**Comment by kimisgold** [comment]
No real reason that these need to be `ul` elements, so I've made them `div`s.
